src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, HTTPException, Response
 from pydantic import BaseModel
-from typing import List
 from src import database as db
-import sqlalchemy
 from sqlalchemy import text
-
 router = APIRouter(
     prefix="/meal-plan",
     tags=["Meal Plan"],
 )
 
 class MealPlanCreate(BaseModel):
-    user_id: int
-    start_date: str
-    end_date: str
-    daily_calorie_goal: int
+    account_id: int
+    name: str
+    description: str
 
 class MealLog(BaseModel):
     date: str
     recipe_id: int
     quantity: int
     meal_type: str  
 
-@router.post("/meal-plans")
+@router.post("/")
 def create_meal_plan(plan: MealPlanCreate):
-    with db.engine.begin() as connection:
-        connection.execution_options(isolation_level="SERIALIZABLE")
-        result = connection.execute(
-            text("""
-                INSERT INTO meal_plans (user_id, start_date, end_date, daily_calorie_goal)
-                VALUES (:user_id, :start_date, :end_date, :daily_calorie_goal)
-                RETURNING id
-            """),
-            {
-                "user_id": plan.user_id,
-                "start_date": plan.start_date,
-                "end_date": plan.end_date,
-                "daily_calorie_goal": plan.daily_calorie_goal
-            }
-        )
-        meal_plan_id = result.scalar()
-    return {"meal_plan_id": meal_plan_id}
+    try:
+        with db.engine.begin() as connection:
+            validate_account = connection.execute(text(
+                    "SELECT id FROM accounts WHERE id = :account_id"),
+                    {"account_id": plan.account_id }
+                ).one_or_none()
 
+            if validate_account is None:
+                raise Exception("Account not found.")
 
-@router.post("/meal-plans/{meal_plan_id}/log")
-def log_meal(meal_plan_id: int, meal: MealLog):
-    with db.engine.begin() as connection:
-        connection.execute(
-            text("""
-                INSERT INTO meal_logs (meal_plan_id, date, recipe_id, quantity, meal_type)
-                VALUES (:meal_plan_id, :date, :recipe_id, :quantity, :meal_type)
-            """),
-            {
-                "meal_plan_id": meal_plan_id,
-                "date": meal.date,
-                "recipe_id": meal.recipe_id,
-                "quantity": meal.quantity,
-                "meal_type": meal.meal_type
-            }
+            result = connection.execute(
+                text("""
+                    INSERT INTO meal_plans (author_id, name, description)
+                    VALUES (:account_id, :name, :description)
+                    RETURNING id
+                """),
+                {
+                    "account_id": plan.account_id,
+                    "name": plan.name,
+                    "description": plan.description
+                }
+            )
+
+            meal_plan_id = result.scalar()
```

Comment:
I think `.scalar()` can return None or a single result? Per the documentation:
https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.scalar
Should be changed to `.scalar_one()` if None result isn't allowed. Idk if it's even possible for this to return None, but this will enforce it to not return an invalid id